src/game/Strategic/Town_Militia.cc

```diff
@@ -246,8 +246,16 @@ void StrategicRemoveMilitiaFromSector(INT16 sMapX, INT16 sMapY, UINT8 ubRank, UI
 
 
 // kill pts are (2 * kills) + assists
-UINT8 CheckOneMilitiaForPromotion(INT16 const x, INT16 const y, UINT8 const current_rank, UINT8 kill_points)
+UINT8 CheckOneMilitiaForPromotion(INT16 const x, INT16 const y, UINT8 &current_rank, UINT8 kill_points)
 {
+	// since the awarding is potentially significantly delayed, make sure they
+	// weren't all promoted already by regular training or killed
+	SECTORINFO& si = SectorInfo[SECTOR(x, y)];
+	while (si.ubNumberOfCivsAtLevel[current_rank] == 0) {
```

Comment:
Because this is passed by reference, I think this should have an additional comment. Something like determine minimal applicable rank.

Reply:
is it not clear from the code?

Reply:
I hate it when references are changed without comments. Preferably at the top of the function.

Reply:
Both are already true ... but I'll go add an extra line.